src/internal/cache-client.ts

```diff
@@ -673,27 +676,51 @@ export class CacheClient {
 
   public async listFetch(
     cacheName: string,
-    listName: string
+    listName: string,
+    startIndex?: number,
+    endIndex?: number
```

Comment:
Can you also add a call in the try block to validate the start and end indexes? See @cprice404 's sorted set fetch PR as a reference.